kingfisher_scrapy/extensions.py

```diff
@@ -373,17 +381,29 @@ def item_scraped(self, item, spider):
             # in case of error send info about it to api
             data['errors'] = json.dumps(item['errors'])
 
-        response = self._post("api/v1/create_collection_file", data)
+        def response_callback(response):
+            if not response.ok:
+                spider.logger.warning("Failed to POST create_collection_file. API status code: {}".format(
+                    response.status_code))
 
-        if not response.ok:
-            spider.logger.warning(
-                'Failed to POST create_collection_file. API status code: {}'.format(response.status_code))
+        self._post_async("api/v1/create_collection_file", data, response_callback)
 
-    def _post(self, url, data):
+    def _post_sync(self, url, data):
         """
-        Wrapper around the requests. Add auth if necessary.
+        Wrapper around the requests to api. Adds auth if necessary.
         """
         if self.username and self.password:
             return requests.post("{}/{}".format(self.url, url), json=data, auth=(self.username, self.password))
         else:
             return requests.post("{}/{}".format(self.url, url), json=data)
+
+    def _post_async(self, url, data, callback):
+        """
+        Wrapper around the treq to api. Adds auth if necessary.
+        """
+        if self.username and self.password:
+            request = treq.post("{}/{}".format(self.url, url), json=data, auth=(self.username, self.password))
+        else:
+            request = treq.post("{}/{}".format(self.url, url), json=data)
```

Comment:
Reduce repetition. Can do same for `_post_sync`.

```suggestion
        kwargs = {}
        if self.username and self.password:
            kwargs['auth'] = (self.username, self.password)
        request = treq.post("{}/{}".format(self.url, url), json=data, **kwargs)
```

Reply:
Frankly, not a big fan of kwargs here. In 1:1 cases, I prefer readability before duplication/repetition. But you are the codebase master.